### PR TITLE
perf(torch): drop redundant channels_last contiguous copy in conv input path

### LINKED ISSUE #23279: torch conv does two NHWC<->NCHW copies when one would do
On the torch backend, `Conv` (`keras/src/backend/torch/nn.py`, `conv()`) makes two input copies per forward pass in the default `channels_last` data format: `_transpose_spatial_inputs` permutes NHWC->NCHW and calls a plain `.contiguous()`, then `_maybe_convert_to_channels_last` immediately calls `.contiguous(memory_format=torch.channels_last)` on the result.

The first copy is redundant every single call: a permuted NHWC->NCHW view of a contiguous input is already `channels_last`-contiguous, so the standard-contiguous tensor produced by step one is never consumed -- the very next line reconverts it to `channels_last` anyway. This isn't data-dependent; it's a fixed property of the transpose, so it fires on every conv forward regardless of input.

Fix: collapse the two steps into one -- have `_transpose_spatial_inputs` permute to a view and call `.contiguous(memory_format=torch.channels_last / channels_last_3d)` directly for the 4-D/5-D conv path, and drop the now-redundant separate `_maybe_convert_to_channels_last` call in `conv()`. Makes the common case zero-copy on input.

Part of the per-call Python-dispatch overhead series in #22561. Fixed by #23199.

### PR DESCRIPTION
**Pull request by pctablet505**
Closes #23279

## Summary

In the torch backend's `conv`, a `channels_last` input was converted twice: `_transpose_spatial_inputs` permuted NHWC→NCHW and called `.contiguous()`, and then `_maybe_convert_to_channels_last` converted that result to the channels_last memory format. The first copy is thrown away by the second.

A permuted NHWC→NCHW view of a contiguous tensor already has channels_last strides, so asking for that memory format directly copies nothing. `_transpose_spatial_inputs` gains a `channels_last_memory_format` argument that requests it, and `conv` drops the follow-up conversion for the input.

The argument selects torch's `memory_format` for the result. It does not change the dimension order — `_transpose_spatial_inputs` always returns channels_first, as its docstring says.

## Why it's safe

The destination is the same. Master already ended at a channels_last tensor via `_maybe_convert_to_channels_last`; this reaches it without the intermediate contiguous copy. Outputs are bitwise identical to master across 1D/2D/3D convs on both CPU and CUDA, `valid` and `same` padding.

The `.contiguous()` added for `torch.export` in #22797 is not dropped — the result is still contiguous, in channels_last format rather than the default. Master's own post-conversion tensor was likewise not default-contiguous, so `torch.export` already saw a channels_last tensor here.

Only `conv` opts in. `depthwise_conv` inherits it by delegating to `conv`; `conv_transpose` and the pooling ops still take the default contiguous path.

## Tests

No new tests. The existing backend-independent conv tests cover this: mutating the change (dropping the copy outright, swapping the 2D/3D memory formats, or perturbing the permutation) is caught by `keras/src/ops/nn_test.py` and `keras/src/layers/convolutional/` in every case where it is detectable at all.

## Contributor Agreement

- [x] I am a human, and not a bot.
- [x] I will be responsible for responding to review comments in a timely manner.
- [x] I will work with the maintainers to push this PR forward until submission.

### CODE REVIEW

File: keras/src/backend/torch/nn_test.py [hunk 1]
```diff
@@ -0,0 +1,201 @@
+"""Tests for PyTorch backend nn ops."""
+
+import numpy as np
+import pytest
```
Comment (hertschuh) [inline keras/src/backend/torch/nn_test.py]: Remove these tests. This should already be covered by the backend independent tests that exist today.

File: keras/src/backend/torch/nn.py [hunk 2]
```diff
@@ -283,21 +283,31 @@ def _apply_same_padding(
     return tnn.pad(inputs, pad=tuple(flattened_padding), mode=mode), 0
 
 
-def _transpose_spatial_inputs(inputs):
+def _transpose_spatial_inputs(inputs, channels_last_format=False):
     """Transpose inputs from channels_last to channels_first format."""
```
Comment (hertschuh) [inline keras/src/backend/torch/nn.py:288]: Something is super confusing here.

Per documentation, `_transpose_spatial_inputs` really means "Transpose from channel last to channel first format". So the output is always "channels first".

The opposite transformation is `_transpose_spatial_outputs`, which really means "Transpose from channel first to channel last format".

So:
- what does the `channels_last_format` argument even mean since the output is always channel first?
- why is `conv` the only one to use this argument and not the other ops?

**Review by hertschuh — CHANGES_REQUESTED** [review]

File: keras/src/backend/torch/nn_test.py (hunk 1 above)
Comment (hertschuh) [reply keras/src/backend/torch/nn_test.py]: Please address

File: keras/src/backend/torch/nn.py (hunk 2 above)
Comment (hertschuh) [reply keras/src/backend/torch/nn.py:288]: Can you answer this question?

Comment (pctablet505) [reply keras/src/backend/torch/nn.py]: **The argument name was wrong — you're right.**

It never changed the order of the dimensions. The output is always channels_first, exactly as the docstring says. All the flag does is choose which memory layout torch uses for the result. I've renamed it to `channels_last_memory_format`.

**Why this saves work**

After the permute, the tensor already has the layout we want. Master then copied it twice to arrive back at that same layout.

Input `(2, 8, 9, 3)`:

| step | stride | copies data? |
|---|---|---|
| input | (216, 27, 3, 1) | — |
| after `permute` | (216, 1, 27, 3) | no |
| master, copy 1 | (216, 72, 9, 1) | yes |
| master, copy 2 | (216, 1, 27, 3) | yes |
| this PR | (216, 1, 27, 3) | no |

Rows 2 and 5 are the same, so both copies were avoidable. The PR just asks for that layout directly.

Results are bitwise identical to master: 1D, 2D and 3D convs, on CPU and CUDA, with `valid` and `same` padding.

**Why only `conv`**

`depthwise_conv` gets it too, because it calls `conv`. I left out `conv_transpose` and pooling on purpose. Pooling would benefit, but `conv_transpose` ends up on a different kernel and its results move slightly (around 1e-2), which would mean changing test tolerances. That belongs in its own PR.

File: keras/src/backend/torch/nn_test.py (hunk 1 above)
Comment (pctablet505) [reply keras/src/backend/torch/nn_test.py]: removed it

**Review by hertschuh — APPROVED** [review]
Thanks for the optimization!
